libsql-server/src/hrana/ws/mod.rs

```diff
@@ -22,7 +22,7 @@ mod session;
 
 struct Server<F: MakeNamespace> {
     namespaces: NamespaceStore<F>,
-    auth: Arc<Auth>,
+    user_auth_strategy: Arc<Box<dyn UserAuthStrategy>>,
```

Comment:
An idea: We could potentially replace `Arc<Auth>` with `Auth` that internally holds an `Arc<dyn UserAuthStrategy>` that way we don't leak the internal implementation details (and there are no generics). This I think would then mean the rest of the code base just gets a `Arc<Auth>` -> `Auth` conversion and it should be really clean.